File: backend/project/tests/values/payment_details.py
class IPaymentDetails():
    def __init__(self, description):
        self.description = description


class CashDetails(IPaymentDetails):
    def __init__(self, description=None):
        super().__init__(description)
        self.payment_method = 'Cash'


class PayPalDetails(IPaymentDetails):
    def __init__(self, description=None, paypal_email=None, reason=None,
                 message=None):
        super().__init__(description)
        self.payment_method = 'PayPal'
        self.paypal_email = paypal_email
        self.reason = reason
        self.message = message


class BankTransfer(IPaymentDetails):
    def __init__(self, description=None, bank_name=None, account_name=None,
                 bsb_number=None, account_number=None):
        super().__init__(description)
        self.payment_method = 'Bank Transfer'
        self.bank_name = bank_name
        self.account_name = account_name
        self.bsb_number = bsb_number
        self.account_number = account_number
        

File: angular-frontend/src/app/models/payment-details.ts
export interface IPaymentDetails {
  payment_method: string;
  description: string;
}

export class BankDetails implements IPaymentDetails {
  payment_method: string = 'Bank Transfer';
  bank_name: string;
  account_name: string;
  bsb_number: string;
  account_number: string;
  description: string;
}

export class PayPalDetails implements IPaymentDetails {
  payment_method: string = 'PayPal';
  email: string;
  reason: string;
  message: string;
  description: string;
}

export class CashDetails implements IPaymentDetails {
  payment_method: string = 'Cash';
  description: string;
}

export class PaymentMethodSelector {
  private validPaymentDetails = [
    new BankDetails(),
    new PayPalDetails(),
    new CashDetails()
  ];
  selectedPaymentMethod: string;

  constructor(public paymentDetailsObjs = new Map<number, IPaymentDetails>()) {
    for (let i = 0; i < this.validPaymentDetails.length; i++) {
      this.paymentDetailsObjs[
        this.validPaymentDetails[i].payment_method
      ] = this.validPaymentDetails[i].payment_method;
    }
  }

  changeSelectedPaymentMethod(name: string) {
    this.selectedPaymentMethod = name;
  }

  getSelectedPaymentMethod(): IPaymentDetails {
    return this.paymentDetailsObjs[this.selectedPaymentMethod]
  }
}